ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -323,23 +323,36 @@ PrjFS_Result PrjFS_WritePlaceholderFile(
         return PrjFS_Result_EInvalidArgs;
     }
     
+    PrjFS_Result result = PrjFS_Result_Invalid;
     PrjFSFileXAttrData fileXattrData = {};
     
     char fullPath[PrjFSMaxPath];
     CombinePaths(s_virtualizationRootFullPath.c_str(), relativePath, fullPath);
     
-    // Mode "wbx" means
-    //  - Create an empty file if none exists
-    //  - Fail if a file already exists at this path
-    FILE* file = fopen(fullPath, "wbx");
+    // Mode "wx" means:
```

Comment:
> Why did you remove the "b" here?

When I was digging into the documentation for `fopen` I found the "b" wasn't doing anything.

From the Mac manpage for `fopen`:

> The mode string can also include the letter "b" after either the "+'' or the first letter.  This is strictly for compatibility with ISO/IEC 9899:1990 ('ISO C90') and has effect only for fmemopen() ; otherwise "b'' is ignored.

The [linux documentation](https://linux.die.net/man/3/fopen) had a few more details:

> The mode string can also include the letter 'b' either as a last character or as a character between the characters in any of the two-character strings described above. This is strictly for compatibility with C89 and has no effect; the 'b' is ignored on all POSIX conforming systems, including Linux. (Other systems may treat text files and binary files differently, and adding the 'b' may be a good idea if you do I/O to a binary file and expect that your program may be ported to non-UNIX environments.)

From this it sounds like C's `fopen` has "b" to support non-POSIX systems, but that on Mac it's not doing anything.

Let me know if you'd rather I leave it in as it doesn't appear to be hurting anything.

Reply:
Got it, I was looking at more generic docs that said that if you don't specify "b", it can translate some of the characters as you read/write, which we don't want. If it's a noop on Mac, then this is fine.